pufferlib/ocean/trash_pickup/trash_pickup.h

```diff
@@ -0,0 +1,596 @@
+
+        // Add the observing agent's own position and carrying status
+        obs[obs_index++] = (float) (env->agent_positions[agent_idx][0]) * norm_factor;
+        obs[obs_index++] = (float) (env->agent_positions[agent_idx][1]) * norm_factor;
+        obs[obs_index++] = env->agent_carrying[agent_idx] ? 1.0f : 0.0f;
+
+        // Add positions of other agents
+        for (int other_idx = 0; other_idx < env->num_agents; other_idx++) {
+            if (other_idx == agent_idx) continue; // Skip self
+            obs[obs_index++] = (float) (env->agent_positions[other_idx][0]) * norm_factor;
+            obs[obs_index++] = (float) (env->agent_positions[other_idx][1]) * norm_factor;
+            obs[obs_index++] = env->agent_carrying[other_idx] ? 1.0f : 0.0f;
+        }
+
+        // Add shared trash data
+        for (int i = 0; i < env->num_trash; i++) {
+            if (env->trash_presence[i]) {
+                obs[obs_index++] = (float) (env->trash_positions[i][0]) * norm_factor;
+                obs[obs_index++] = (float) (env->trash_positions[i][1]) * norm_factor;
+                obs[obs_index++] = 1.0f; // Trash present
+            } else {
+                obs[obs_index++] = -1.0f; // Placeholder for x
+                obs[obs_index++] = -1.0f; // Placeholder for y
+                obs[obs_index++] = 0.0f; // Trash not present
+            }
+        }
+
+        // Add shared bin data
+        for (int i = 0; i < env->num_bins; i++) {
+            obs[obs_index++] = env->bin_positions[i][0] * norm_factor;
+            obs[obs_index++] = env->bin_positions[i][1] * norm_factor;
+        }
+    }
+}
+
+// Helper functions
+void place_random_items(CTrashPickupEnv* env, int count, int item_type) {
+    int placed = 0;
+    while (placed < count) 
+    {
+        int x = rand() % env->grid_size;
+        int y = rand() % env->grid_size;
+
+        if (env->grid[INDEX(env, x, y)].type == EMPTY) {
+            env->grid[INDEX(env, x, y)].type = item_type;
+
+            if (item_type == AGENT) {
+                env->grid[INDEX(env, x, y)].index = placed;
+                env->agent_positions[placed][0] = x;
+                env->agent_positions[placed][1] = y;
+            } else if (item_type == TRASH) {
+                env->grid[INDEX(env, x, y)].index = placed; // Set trash index
+                env->trash_positions[placed][0] = x;
+                env->trash_positions[placed][1] = y;
+                env->trash_presence[placed] = 1;
+            } else if (item_type == TRASH_BIN) {
+                env->grid[INDEX(env, x, y)].index = placed; // Set bin index
+                env->bin_positions[placed][0] = x;
+                env->bin_positions[placed][1] = y;
+            }
+
+            placed++;
+        }
+    }
+}
+
+void add_reward(CTrashPickupEnv* env, float reward){
+    // Reward all agents to promote cooperative behavior rather than competitive
+    for (int i = 0; i < env->num_agents; i++){
```

Comment:
I guess if you have 3 agents, this MIGHT be reasonable. Naive sharing scales absolutely terribly for more than a few agents. At the very least, you want to lerp personal reward vs. team reward

Reply:
I'd just do individual reward for now until you see it doesn't work

Reply:
Fixed (to be pushed).